packages/calcite-components/src/components/dialog/dialog.scss

```diff
@@ -0,0 +1,237 @@
+ * @prop --calcite-dialog-width: Specifies the width of the component, using `px`, `em`, `rem`, `vw`, or `%`. Will never exceed the width of the viewport. Will not apply if `placement="cover"` if set.
+ * @prop --calcite-dialog-height: Specifies the height of the component, using `px`, `em`, `rem`, `vh`, or `%`. Will never exceed the height of the viewport. Will not apply if `placement="cover"` if set.
+ * @prop --calcite-dialog-panel-footer-padding: Specifies the padding of the component's footer.
+ * @prop --calcite-dialog-panel-header-border-block-end: Specifies the component header's block end border.
+ */
+
+:host {
+  --calcite-dialog-scrim-background: #{rgba($calcite-color-neutral-blk-240, $calcite-opacity-85)};
+  @apply flex inset-0 z-overlay pointer-events-none;
+  // the dialog should always use a dark scrim, regardless of light / dark mode - matches value in global.scss
+  --calcite-dialog-scrim-background-internal: #{rgba($calcite-color-neutral-blk-240, $calcite-opacity-85)};
+}
+
+:host([modal]) {
+  @apply absolute;
+}
+
+.container {
+  @apply text-color-2 pointer-events-auto
+    fixed
+    inset-0
+    flex
+    items-center
+    justify-center
+    overflow-y-hidden
+    opacity-0
+    z-overlay;
+  visibility: hidden;
+  transition:
+    visibility 0ms linear var(--calcite-internal-animation-timing-slow),
+    opacity var(--calcite-internal-animation-timing-slow) $easing-function;
+}
+
+:host([placement="top"]) .container {
+  @apply justify-center items-start;
+}
+
+:host([placement="top-start"]) .container {
+  @apply justify-start items-start;
+}
+
+:host([placement="top-end"]) .container {
+  @apply justify-end items-start;
+}
+
+:host([placement="bottom"]) .container {
+  @apply justify-center items-end;
+}
+
+:host([placement="bottom-start"]) .container {
+  @apply justify-start items-end;
+}
+
+:host([placement="bottom-end"]) .container {
+  @apply justify-end items-end;
+}
+
+:host(:not([modal])) .container {
+  @apply pointer-events-none;
+}
+
+:host([scale="s"]) {
+  --calcite-dialog-padding-internal: theme("spacing.3");
+  --calcite-dialog-padding-large-internal: theme("spacing.4");
+  --calcite-dialog-title-text-internal: theme("fontSize.1h");
+  --calcite-dialog-content-text-internal: theme("fontSize.n1");
+}
+
+:host([scale="m"]) {
+  --calcite-dialog-padding-internal: theme("spacing.4");
+  --calcite-dialog-padding-large-internal: theme("spacing.5");
+  --calcite-dialog-title-text-internal: theme("fontSize.2h");
+  --calcite-dialog-content-text-internal: theme("fontSize.0");
+}
+
+:host([scale="l"]) {
+  --calcite-dialog-padding-internal: theme("spacing.5");
+  --calcite-dialog-padding-large-internal: theme("spacing.6");
+  --calcite-dialog-title-text-internal: theme("fontSize.3h");
+  --calcite-dialog-content-text-internal: theme("fontSize.1");
+}
+
+.scrim {
+  --calcite-scrim-background: var(--calcite-dialog-scrim-background, var(--calcite-color-transparent-scrim));
```

Comment:
can you add
`--calcite-scrim-background-color: var(--calcite-dialog-scrim-background-color, var(--calcite-color-transparent-scrim));`
to make merging with the token's epic easier?

Reply:
ofcourse!

Reply:
Note: we have to have both of these vars for now as the new one doesn't work yet.